doc/adr/0004-align-technology-practice-process.md

```diff
@@ -0,0 +1,45 @@
+## 4. Align technology, practice and process across the Libero Suite where appropriate
+
+Date: 2019-10-10
+
+### Status
+
+Accepted
+
+### Context
+
+The Libero suite of products currently contains four products: Reviewer, Producer, Publisher and Data Hub. All have different problems to solve where different technologies, frameworks, libraries, concepts and practices help in different ways. For example, Publisher takes a large amount of traffic from a globally diverse population who generally drop in quickly to read content and then disappear. In contrast, Reviewer serves users that are committed to filling forms and uploading content, and has a fraction of the traffic compared to Publisher. Striking the right balance between addressing these different needs while promoting a good experience for developers and service providers is the key to our alignment strategy.
+
+There are a number of personas that we need to balance:
+
+- Web site / app end users (readers, authors, editors and peer reviewers)
+- Publishing staff (staff at organisations using the software)
+- Community developers who wish to extend and configure the products
+- Core developers (all at eLife currently) who may want to work across products and share services but also retain autonomy for a given product by reducing coupling
+- Service providers who want to configure and deploy instances of the products
+
+### Decision
+
+The approach is to align in a number of well-defined areas and record this in ADRs. However, strong alignment is not a requirement where it is detrimental to the way the product solves the problems for its given use cases or where it creates undesirable coupling between teams (resulting in synchronisation overheads and reduced autonomy/empowerment). For example, using the same framework for Publisher and Reviewer on the front-end would mean that either single-page performance or interactivity, respectively, would have to suffer, so a better choice would be weaker alignment in that area, as Publisher requires the first and Reviewer depends on the latter.
+
+The process is to explore every reasonable opportunity for a strong alignment, and to pursue it in practice when it becomes apparent that its implementation will not be detrimental to the web apps’ end user experience. Therefore we are not seeking alignment for alignment’s sake, nor are we ignoring it for the sake of the web apps’ end users.
```

Comment:
this paragraph seems to suggest always to explore a strong alignment first, and pursue if it's not detrimental; but the list of questions say the first thought is consider whether there is positive value in the alignment?

Reply:
Good observation. The intent is to explore strong alignment but then ensure we have considered and record the value during the discovery - hence the need for the question. Hopefully, most of the time, the value will be easily obvious ;) The inclusion of the question on the potential diversion prompts us to consider this from either starting point. Following all the questions should get us to the result we want.